modules/oe_content_event/src/Plugin/InternalLinkSourceFilter/EventLinkSourceFilter.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_event\Plugin\InternalLinkSourceFilter;
+
+use Drupal\Core\Cache\RefinableCacheableDependencyInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\oe_link_lists_internal_source\InternalLinkSourceFilterInterface;
+use Drupal\oe_link_lists_internal_source\InternalLinkSourceFilterPluginBase;
+
+/**
+ * Event link source filter class.
+ *
+ * @InternalLinkSourceFilter(
+ *   id = "event_link_source_filter",
+ *   label = @Translation("Event Link Source Filter"),
+ *   description = @Translation("Filters for link lists using events as the internal source."),
+ *   entity_types = {
+ *     "node" = {
+ *       "oe_event",
+ *     },
+ *   },
+ * )
+ */
+class EventLinkSourceFilter extends InternalLinkSourceFilterPluginBase implements InternalLinkSourceFilterInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'time' => 'upcoming',
```

Comment:
As an option, it could be class constants UPCOMING and PAST with values 1 and -1 accordingly.

Reply:
Agreed.